lms/djangoapps/program_enrollments/rest_api/v1/views.py

```diff
@@ -709,15 +709,17 @@ def get(self, request):
 
         if request_user.is_staff:
             programs = get_programs_by_type(request.site, requested_program_type)
-        elif self.is_course_staff(request_user):
-            programs = self.get_programs_user_is_course_staff_for(request_user, requested_program_type)
         else:
-            program_enrollments = fetch_program_enrollments_by_student(
-                user=request.user,
-                program_enrollment_statuses=ProgramEnrollmentStatuses.__ACTIVE__,
-            )
-            uuids = [enrollment.program_uuid for enrollment in program_enrollments]
-            programs = get_programs(uuids=uuids) or []
+            # Check if the user is a course staff of any course which is a part of a program.
+            programs = self.get_programs_user_is_course_staff_for(request_user, requested_program_type)
+            if not programs:
```

Comment:
I might be missing something here, but don't we want to include both programs the user is course staff for and the programs they are enrolled in as a learner?  This looks like it will only return one or the other still

Reply:
Yes, you are missing something. The problem is that, the condition `self.is_course_staff(request_user)` returns true when the user is course staff of a course that's not part of the program. That's not the intension of the check. The intention of the check is to figure out if the user is a course_staff of any course, which is a part of any program. That is the situation returned by the `self.get_programs_user_is_course_staff_for` function. If that function returns empty. We know the learner is not a course staff, then we can proceed to the student logic below. To sum up, we really should not rely on the `is_course_staff` logic. Hence that function is removed.

Reply:
What if a learner is considered 'course_staff' for program A, but enrolled as a learner in program B.  Assuming both of those programs are master's programs, won't the learner have the same issue where program B won't return even though they have read access?

Reply:
@Zacharis278 Interesting scenario. We haven't encountered that on production yet. But I think it's valuable to include this scenario. I'll talk with Product a bit and decide.

Reply:
@Zacharis278 OK, I made another commit to take care of the scenario you described. Please review again.